cmd/clusterctl/client/repository/components.go

```diff
@@ -251,6 +255,12 @@ func NewComponents(input ComponentsInput) (Components, error) {
 	// Add common labels.
 	objs = addCommonLabels(objs, input.Provider)
 
+	if input.ObjModifier != nil {
+		objs, err = input.ObjModifier(objs)
```

Comment:
I'm not sure what is the exact use case for this modifier, but won't be more explicit if we create a new ComponentObject starting from the one returned by this func (instead of plugging modification inside NewComponents)

Reply:
hi @fabriziopandini it is used by the operator to implement the ManagerSpec and Deployment changes, see here https://github.com/kubernetes-sigs/cluster-api/pull/4926/files#diff-c1e3500c1f99c1f9965fdc568c7a6d0b6bb1ab34748829c19e481ad19d704437R143
If I have 
```
    s.components, _ = repository.NewComponents(repository.ComponentsInput{},...)
    modifiedIObjs, _ = customizeObjectsFn(provider, s.componets.Objs())
    s.components.SetObjs(modifiedIObjs)
```
I can't modify the "objs" as they are private, so to do as you suggest we will need a SetObjs() in the Components interface.

Do you prefer this approach?

Reply:
It seems the apidiff doesn't like this.. 
```
   Incompatible changes:
  - Components.SetObjs: added 
```

Reply:
@fabriziopandini thanks, updated with that idea.